### ISSUE TITLE
Untruncated long filenames break post style

### ISSUE THREAD

**Issue by nebukazar** [ccd0/4chan-x#1082]
I was about to fix this in Oneechan (https://github.com/nebukazar/OneeChan/issues/90) when I noticed it happens in 4chan X as well:

![capture](https://cloud.githubusercontent.com/assets/5222254/18571185/621167d2-7ba8-11e6-889f-4dc750c00035.PNG)

**Comment by ccd0** [comment]
Various other stuff causes this problem, including long links and long TeX formulas on /sci/. A lot of this happens in native 4chan as well. Would be nice to figure out a good general solution.

see also #467, #765

I had another shot at this, and although I did have luck with setting a `max-width` and turning on `word-wrap: break-word`, I didn't come up with a good way of figuring out what the max width would be in general. The `table-cell` approach from #467 might also be something to consider.

**Comment by nebukazar** [comment]
What if we put a `word-break: break-word;` on the `body` ? Is this too dangerous?

It fixes the example in OP but I wonder if it works for very long on word links or TeX. Any threads to test this you know of? Maybe trying it on /qa/.

**Comment by ccd0** [comment]
`word-break: break-word;` is nonstandard unfortunately. It only works in Webkit/Blink.

**Comment by saxamaphone69** [comment]
I've had
```css
word-wrap: break-word;
overflow-wrap: break-word;
-webkit-hyphens: auto;
   -moz-hyphens: auto;
        hyphens: auto;
```
on `:root` in my userstyle for yonks now.

Don't know if that might help this situation.

**Comment by ccd0** [comment]
@saxamaphone69
Doesn't seem to help in either Firefox or Chromium.

Here's the link in the screenshot for easier testing:
https://boards.4chan.org/g/thread/56582381#p56583069
The effect shown seems to kick in at a browser width of around 750px, although the exact width will presumably vary with font choices.

**Comment by ccd0** [comment]
@saxamaphone69
Of course, if you have a maximum width set somewhere in your CSS, that would explain why it works for you.

**Comment by saxamaphone69** [comment]
Turns out this also doesn't work in my ss16 userstyle, and the hovered file info goes off the page. Heck, I just realised Chrome doesn't even support `hyphens`.

It may not be a permanent solution, but I gave
```css
.file {
  word-break: break-all;
}
```
a shot, and (I feel) it does the job at solving the long filename (tested in Chrome/Firefox).
No overflowing of text and no fast flashing when trying to hover.
You could even target specific browsers with the `.ua-XXX` if need be, I guess.

I had `max-width: 90vw` on `.post` as well at one point, don't know if playing with that might help. Also don't know how that might help long links or formulas.

**Comment by nebukazar** [comment]
Problem with `word-break: break-all` is that it will cut off words arbitrarily instead of preserving the word.

I've played around with it yesterday but didn't find a solution. The solution for Chrome is easy, just use `word-break: break-word;`.

For Firefox, this seems to mitigate the problem but only for non inlined posts:

```css
#delform {
  overflow-wrap: break-word;
}
```
Gud enuff I'd say.

**Comment by ccd0** [comment]
I think I'm just going to put it on the file line for now. 4chan breaks long text/links with the `<wbr>`s, and TeX on /sci/ was either unchanged (`word-break: break-word;` in Chromium) or made worse, cutting off the text (max-width + `overflow-wrap` / `word-wrap: break-word;` in Firefox).

Looks like they're considering implementing `word-break: break-word;` in Firefox and Edge for web compatibility reasons and adding it to the spec.
https://lists.w3.org/Archives/Public/www-style/2016Mar/0352.html
https://readable-email.org/list/www-style/topic/css-text-word-break-break-word
https://bugzilla.mozilla.org/show_bug.cgi?id=1296042
